pyspark/dl/optim/optimizer.py

```diff
@@ -140,6 +142,179 @@ class Step(JavaValue):
+    :param weightdecays 1D tensor of individual weight decays
+    >>> sgd = SGD()
+    creating: createDefault
+    creating: createSGD
+    """
+    def __init__(self,
+                 learningrate=1e-3,
+                 learningrate_decay=0.0,
+                 weightdecay=0.0,
+                 momentum=0.0,
+                 dampening=DOUBLEMAX,
+                 nesterov=False,
+                 leaningrate_schedule=None,
+                 learningrates=None,
+                 weightdecays=None,
+                 bigdl_type="float"):
+        JavaValue.__init__(self, None, bigdl_type, learningrate, learningrate_decay, weightdecay,
+                           momentum, dampening, nesterov,
+                           leaningrate_schedule if (leaningrate_schedule) else Default(),
+                           JTensor.from_ndarray(learningrates), JTensor.from_ndarray(weightdecays))
+
+class Adagrad(JavaValue):
+    """
+    An implementation of Adagrad. See the original paper:
+    http://jmlr.org/papers/volume12/duchi11a/duchi11a.pdf
+
+    :param learningrate learning rate
+    :param learningrate_decay learning rate decay
+    :param weightdecay weight decay
+    >>> adagrad = Adagrad()
+    creating: createAdagrad
+    """
+    def __init__(self,
+                 learningrate=1e-3,
+                 learningrate_decay=0.0,
+                 weightdecay=0.0,
+                 bigdl_type="float"):
+        JavaValue.__init__(self, None, bigdl_type, learningrate, learningrate_decay, weightdecay)
+
+class LBFGS(JavaValue):
+    """
+    This implementation of L-BFGS relies on a user-provided line
+    search function (state.lineSearch). If this function is not
+    provided, then a simple learningRate is used to produce fixed
+    size steps. Fixed size steps are much less costly than line
+    searches, and can be useful for stochastic problems.
+    The learning rate is used even when a line search is provided.
+    This is also useful for large-scale stochastic problems, where
+    opfunc is a noisy approximation of f(x). In that case, the learning
+    rate allows a reduction of confidence in the step size.
+
+    :param max_iter Maximum number of iterations allowed
+    :param max_eval Maximum number of function evaluations
+    :param tolfun Termination tolerance on the first-order optimality
+    :param tolx Termination tol on progress in terms of func/param changes
+    :param ncorrection
+    :param learningrate
+    :param verbose
+    :param linesearch A line search function
+    :param linesearch_options If no line search provided, then a fixed step size is used
+    >>> lbfgs = LBFGS()
+    creating: createLBFGS
+    """
+    def __init__(self,
+                 max_iter=20,
+                 max_eval=DOUBLEMAX,
+                 tolfun=1e-5,
+                 tolx=1e-9,
+                 ncorrection=100,
+                 learningrate=1.0,
+                 verbose=False,
+                 linesearch=None,
```

Comment:
linesearch function is a scala object, but no corresponding python function. Perhaps I can add one.